src/rspamadm/configdump.c

```diff
@@ -305,6 +308,119 @@ rspamadm_configdump (gint argc, gchar **argv, const struct rspamadm_command *cmd
 			exit (EXIT_SUCCESS);
 		}
 
+		if (symbol_full_details) {
+			/*
+			 * Create object from symbols groups and output it using the
+			 * specified format
+			 */
+			ucl_object_t *out = ucl_object_typed_new (UCL_OBJECT);
+			GHashTableIter it;
+			gpointer sk, sv;
+
+			g_hash_table_iter_init (&it, cfg->symbols);
+			ucl_object_t *sym_ucl = ucl_object_typed_new (UCL_OBJECT);
+			const ucl_object_t *all_symbols_ucl = ucl_object_lookup(cfg->rcl_obj, "symbols");
+
+			while (g_hash_table_iter_next (&it, &sk, &sv)) {
+				const gchar *sym_name = (const gchar *)sk;
+				struct rspamd_symbol *s = (struct rspamd_symbol *)sv;
+				ucl_object_t *this_sym_ucl = ucl_object_typed_new (UCL_OBJECT);
+
+				ucl_object_insert_key (this_sym_ucl,
+						ucl_object_fromdouble (s->score),
+						"score", strlen ("score"),
+						false);
+				ucl_object_insert_key (this_sym_ucl,
+						ucl_object_fromstring (s->description),
```

Comment:
`s->description` might be NULL...

Reply:
The UCL object code will handle that based on what I read there... ucl_object_fromstring() returns NULL if s->description is NULL, and then ucl_object_insert_key does not insert anything (and return false in that scenario).  In my local testing, I verified this works as expected (and no NULL ptr dereferences) for symbols without descriptions.